File: app/src/main/java/com/example/reminderapp/CallReminderReceiver.java
package com.example.reminderapp;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class CallReminderReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        String phoneNumber = intent.getStringExtra("phone");
        if (phoneNumber == null) phoneNumber = "";

        if (NotificationHelper.NORMAL_CALL_ACTION.equals(action)) {
            CallHelper.makeNormalCall(context, phoneNumber);
        } else if (NotificationHelper.ONLINE_CALL_ACTION.equals(action)) {
            CallHelper.makeWhatsAppCall(context, phoneNumber);
        } else {
            // This is the alarm trigger
            int id = intent.getIntExtra("id", 0);
            String name = intent.getStringExtra("name");
            if (name == null) name = "Contact";
            String notes = intent.getStringExtra("notes");
            NotificationHelper.showCallNotification(context, id, name, phoneNumber, notes);
        }
    }
}

File: app/src/main/java/com/example/reminderapp/CallReminderReceiver.kt
package com.example.reminderapp

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class CallReminderReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        val phoneNumber = intent.getStringExtra("phone") ?: ""

        when (action) {
            NotificationHelper.NORMAL_CALL_ACTION -> {
                CallHelper.makeNormalCall(context, phoneNumber)
            }
            NotificationHelper.ONLINE_CALL_ACTION -> {
                CallHelper.makeWhatsAppCall(context, phoneNumber)
            }
            else -> {
                // This is the alarm trigger
                val id = intent.getIntExtra("id", 0)
                val name = intent.getStringExtra("name") ?: "Contact"
                val notes = intent.getStringExtra("notes")
                NotificationHelper.showCallNotification(context, id, name, phoneNumber, notes)
            }
        }
    }
}